### PR TITLE
Multi from aliases

### LINKED ISSUE #188: Query including Distinct not working
```
sql_new="SELECT DISTINCT customer_id FROM table"

parsed = Parser(sql_new)

print("columns-->", parsed.columns)
```
Result:
`columns--> []`

The above query is not working .

### PR DESCRIPTION
**Pull request by collerek**
Fix for from clause in extract function close: #186
Fix for wrong alias check, close #188

Some simplifications:
- Added `TokenType` enum and `token_type` to token and populate it with enum value
- Merged token values with tables and database delimiter into one token and removed left_expand token property
- Cleaned no longer required checks for `is_dot` property as there are no dot tokens anymore
- Moved populating `column_aliases_dict` dictionary to `column_aliases_names`
- Added flake8 to warn about complexity before CodeFactor - also split long string lines and remove trailing whitespace

### CODE REVIEW

**Comment by macbre** [comment]
It seems that `flake8` does not work too well with `black` formatting. See the failing CI check above.

Can we move this commit (c970453) to a separate PR and deal with `flake8` separately?

**Comment by collerek** [comment]
I forgot to run black before commit 🤦

**Comment by macbre** [comment]
@collerek - can you resolve the poetry.lock conflict?

**Comment by collerek** [comment]
@macbre - sure, resolved. 
Also simplified the `tokens` to fix CodeFactor error.

File: sql_metadata/parser.py [hunk 1]
```diff
@@ -133,17 +133,35 @@ def tokens(self) -> List[SQLToken]:
...
+                    value = f"{prev_value}.{value}"
+                token.value = value
+                combine_flag = False
+
             if index > 0:
                 # create links between consecutive tokens
-                token.previous_token = tokens[index - 1]
-                tokens[index - 1].next_token = token
+                token.previous_token = tokens[-1]
+                tokens[-1].next_token = token
```
Comment (macbre) [inline sql_metadata/parser.py]: Nice optimization 🙂

File: sql_metadata/parser.py [hunk 2]
```diff
@@ -154,13 +154,9 @@ def tokens(self) -> List[SQLToken]:
...
-                value = token.value
-                prev_value = non_empty_tokens[index - 2].value.strip("`").strip('"')
-                value = f"{prev_value}.{value}"
-                if index >= 3 and str(non_empty_tokens[index - 3]) == ".":
-                    prev_value = non_empty_tokens[index - 4].value.strip("`").strip('"')
-                    value = f"{prev_value}.{value}"
-                token.value = value
+                self._combine_qualified_names(
+                    index=index, token=token, non_empty_tokens=non_empty_tokens
+                )
```
Comment (macbre) [inline sql_metadata/parser.py]: 👌🏻

**Review by macbre — APPROVED** [review]

**Comment by macbre** [comment]
@collerek - looks great. Please resolve Poetry files conflicts and we're good to go.

**Pull request merged**